src/CompilerPass/AuthenticationManagerPolyfillPass.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/*
+
+This file was created by developers working at BitBag
+
+Do you need more information about us and what we do? Visit our   website!
+
+We are hiring developers from all over the world. Join us and start your new, exciting adventure and become part of us: https://bitbag.io/career
+*/
+declare(strict_types=1);
+
+namespace BitBag\SyliusCrossSellingPlugin\CompilerPass;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+final class AuthenticationManagerPolyfillPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        if (
+            false === $container->has('security.authentication_manager')
+            &&
+            true === $container->has('security.authentication.manager')
+        ) {
+            $container->setAlias('security.authentication_manager', 'security.authentication.manager');
```

Comment:
I'd say that this one is wrong, you're basically changing "core" service id/alias.
My suggestion is to create an bundle-wise alias for service (i.e. `bitbag_cross_selling.security.authentication_manager`) that will be an alias for any of mentioned above, and then you perform this "switch" inside of CompilerPass.